vignettes/GetStarted.Rmd

```diff
@@ -0,0 +1,100 @@
+---
+title: "Get started with the bayesian package"
+author: "Paul-Christian Bürkner & Hamada S. Badr"
+date: "`r Sys.Date()`"
+output:
+  rmarkdown::html_vignette:
+    keep_md: true
+vignette: >
+  %\VignetteIndexEntry{Getting Started}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r}
+library(recipes)
+library(workflows)
+library(bayesian)
+```
+
+As a simple example, we will model the seizure counts in epileptic patients to
+investigate whether the treatment (represented by variable `Trt`) can reduce the
+seizure counts and whether the effect of the treatment varies with the baseline
+number of seizures a person had before treatment (variable `Base`) and with the
+age of the person (variable `Age)`. As we have multiple observations per `person`,
+a group-level intercept is incorporated to account for the resulting dependency
+in the data. In a first step, we use the `recipes` package to prepare (a recipe
+for) the `epilepsy` data. This data set is shipped with the `brms` package,
+which is automatically loaded by `bayesian`.
+
+```{r}
+epi_recipe <- epilepsy %>% 
+  recipe() %>%
+  update_role(count, new_role = "outcome") %>%
+  update_role(Trt, Age, Base, patient, new_role = "predictor") %>%
+  update_role(patient) %>%
+  step_normalize(Age, Base)
+```
+
+```{r}
+print(epi_recipe)
+```
+
+Above, we not only define the roles of the relevant variables but also 
+normalized the `Age` and `Base` predictors to facilitate model fitting later
+on. In the next step, we use `bayesian` to set up a basic model structure.
+
+```{r}
+epi_model <- bayesian() %>%
+  set_engine("brms") %>%
+  set_mode("regression") %>%
+  update(family = poisson())
```

Comment:
Why do you want to define the arguments by `update` not inside `bayesian`? This means that you specify the default arguments and then update them other than specifying the model at the first call. Both should be identical,

Reply:
True but I want to show users how they can later on update an existing model by overwriting only specific arguments. I am fine with either option of course.

Reply:
Sounds good. I like the idea of showing the case for `update` but the users might think it should be applied everytime they build a model. I think we could specify the model with `bayesian(family = poisson())` and show the users how to change/update the `family` in the following step.

Reply:
Ok. changed it accordingly.